Modules/Loadable/VolumeRendering/MRMLDM/vtkMRMLVolumeRenderingDisplayableManager.h

```diff
@@ -65,6 +65,15 @@ class VTK_SLICER_VOLUMERENDERING_MODULE_MRMLDISPLAYABLEMANAGER_EXPORT vtkMRMLVol
   /// Get the MRML ID of the picked node, returns empty string if no pick
   const char* GetPickedNodeID() override;
 
+  /// @{
+  /// Experimental function to adjust the maximum 3D texture size.
+  /// Along each axis, the volume will be split up so that each partition is not larger than this maximum size.
+  /// Default is 2048 on macOS (as Apple hardware typically is limited to this maximum texture size)
```

Comment:
Where do you go to find that 2048 is the maximum texture on your Apple hardware?

Reply:
It is based on experimenting on Apple user's computers with the maximum volume size that can be rendered without setting partitions (and then seeing that setting partition>1 fixes the rendering problem). If anyone shows up with an Apple computer where a higher value is supported then we can increase the value.

Reply:
It looks like the value of 2048 matches what is listed for “Maximum 3D texture width, height, and depth”

https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf

Reply:
![image](https://github.com/user-attachments/assets/1c55f36e-c0e6-4f77-8dc6-99c182625d30)

Reply:
Nice find! It is possible in theory that Apple will increase this value, but most likely they will not touch their OpenGL support. Also, VTK's new webGPU backend may implement automatic partitioning. So, the current implementation would be needed probably only for a couple of years.